Sources/NIOHTTP2/GlitchesMonitor.swift

```diff
@@ -0,0 +1,71 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2025 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+package struct GlitchesMonitor {
```

Comment:
Are we sure `package` visibility is a good idea here? It enables release mode unit tests, sure, but it also causes the optimizer to generate suboptimal code. Given that swift-nio-http2 uses `@testable import` extensively _already_, I don't think we gain much by avoiding it here.

Reply:
It's suboptimal because it inhibits some optimizations that the compiler can make by virtue of being able to observe all usage sites. An `internal` symbol or type, except when compiled with `-enable-testability`, has all usage sites visible in the code. This can allow strong effects to be inferred, some types to be deleted, etc.

`package` visibility doesn't do that: it's essentially the same visibility as `-enable-testability` would trigger, but all the time. Its saving grace is that it works equivalently well in both modes.

I think `package` should be preferred in contexts where running the unit tests in release mode is meaningful. I personally don't care much about those, but users who do should go for it. Otherwise, I'd recommend sticking with `@testable import`.